extensions/control-plane/provision/provision-oauth2/README.md

```diff
@@ -0,0 +1,18 @@
+# OAuth2 Provision
+
+Permits to attach an OAuth2 token to the transfer, needed when the source or the sink require it.
+
+## How it works
+![](docs/diagram.png)
+
+The token will be requested in the provision phase and stored in the vault, then it will be used by the `data-plane-http`
```

Comment:
If I'm honest, I would improve the README in the following aspects:
- In what situation would one use this extension? Meaning: is it used to authenticate the provider against a _private_ API in his own environment, or is it used to authenticate a consumer against a publicly exposed API of the provider (i.e. the proxy usecase)?
- To me it is not clear, whether the `consumer` or the `provider` (or both?) have to request the OAuth token. In any case, it would be beneficial to clarify this, in the diagram and/or the README.
- Which Oauth2 flows does this apply to? And if it applies to all, or if the flow is irrelevant to us ("out-of-band"), that should be noted too IMO.
- What happens if the token expires between the provisioning and the transfer phase? That could happen if more resources are to be provisioned, and one of them takes longer than the expiry time of the token (e.g. auth code flow)?